swarm/network/kademlia.go

```diff
@@ -836,7 +836,7 @@ func (k *Kademlia) GetHealthInfo(pp *PeerPot) *Health {
 
 	// check saturation
 	unsaturatedBins := k.getUnsaturatedBins(pp.PeersPerBin, depth)
-	saturated := len(unsaturatedBins) == 0 && depth == len(pp.PeersPerBin)
+	saturated := depth == len(pp.PeersPerBin) && len(unsaturatedBins) == 0
```

Comment:
can we please just put this check in the function, ie. return immediately if it is false?

Reply:
Why return an error and not simply bool? There is nothing which can cause an error here.

It is either saturated or not. That is what is currently implemented